src/Microsoft.AspNet.OData.Shared/DeltaOfTStructuralType.cs

```diff
@@ -736,6 +779,60 @@ private void CopyChangedDynamicValues(TStructuralType targetEntity)
                 targetEntity);
         }
 
+        /// <summary>
+        /// Copies changed nested properties and leaves the unchanged properties
+        /// </summary>
+        /// <param name="original">The structural object</param>
+        private void CopyChangedNestedProperties(TStructuralType original)
+        {
+            // For nested resources.
+            foreach (string nestedResourceName in _deltaNestedResources.Keys)
+            {
+                // Patch for each nested resource changed under this TStructuralType.
+                dynamic deltaNestedResource = _deltaNestedResources[nestedResourceName];
+                dynamic originalNestedResource = null;
+
+                if (!TryGetPropertyRef(original, nestedResourceName, out originalNestedResource))
+                {
+                    throw Error.Argument(nestedResourceName, SRResources.DeltaNestedResourceNameNotFound,
+                        nestedResourceName, original.GetType());
+                }
+
+                if (originalNestedResource == null)
+                {
+                    // When patching original target of null value, directly set nested resource.
+                    deltaNestedResource = _deltaNestedResources[nestedResourceName];
```

Comment:
Why do we do this here, yet we already initialized `deltaNestedResource` on line 792?
![image](https://user-images.githubusercontent.com/28291332/188829519-eba85821-9e1f-4f02-8877-3a68d2e9ee49.png)